spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -195,5 +195,10 @@ protected StructLike asStructLike(InternalRow row) {
     protected InputFile getInputFile(String location) {
       return RowDataReader.this.getInputFile(location);
     }
+
+    @Override
+    protected void markRowDeleted(InternalRow row) {
+      row.setBoolean(columnIsDeletedIndex(), true);
+    }
```

Comment:
[question] should we add a precondition here that `columnIsDeletedIndex` should not be -1 ? Your thoughts

presently we have made sure we will invoke markRowDelete only when hasColumnIsDeleted is true implying `columnIsDeletedIndex` can't be ideally -1.

Reply:
Yeah, we don't need to check since the index cannot be -1 if hasColumnIsDeleted is true.